addons/account/models/res_partner_bank.py

```diff
@@ -63,6 +64,21 @@ def _check_allow_out_payment(self):
                 if not self.env.user.has_group('account.group_validate_bank_account'):
                     raise ValidationError(_('You do not have the right to trust or un-trust a bank account.'))
 
+    @api.depends('acc_number')
+    def _compute_alert_duplicate_bank_account(self):
+        """Add a warning if the bank account is already registered with another partner."""
+        duplicates = self.env['res.partner.bank'].search([('acc_number', 'in', self.mapped('acc_number'))]).grouped('acc_number')
+        for bank in self:
+            bank.duplicate_account_warning = False
+            if bank.acc_number:
+                if duplicate := duplicates.get(bank.acc_number):
+                    if bank.id != duplicate.id:
+                        bank.duplicate_account_warning = _(
+                            "Warning : The Bank Account could be a duplicate of %(name)s (<a href=%(url)s>%(partner)s</a>)",
+                            name=duplicate.display_name,
+                            url=duplicate.partner_id.id,
+                            partner=duplicate.partner_id.display_name)
```

Comment:
```suggestion
            if not bank.acc_number:
                continue
            
            duplicate = duplicates.get(bank.acc_number)
            
            if duplicate and bank.id != duplicate.id:
                bank.duplicate_account_warning = _(
                    "Warning : The Bank Account could be a duplicate of %(name)s (<a href=%(url)s>%(partner)s</a>)",
                    name=duplicate.display_name,
                    url=duplicate.partner_id.id,
                    partner=duplicate.partner_id.display_name
                )
```
By using a [guard clause](https://en.wikipedia.org/wiki/Guard_(computer_science)) on the first if statement, we can remove one level. By removing the walrus and reworking the condition we can get rid of another level.

It should do the exact same. I didn't test it and it might not be 100% correct but you get the idea.